### ISSUE TITLE
Eager loading for images

### ISSUE THREAD

**Issue by Matvik** [zxbodya/yii2-gallery-manager#54]
There is a problem - because  "images" is not a relation, we can not apply eager loading, for example, when displaying several products with their images. So, there will be a number of database queries executed. It is not good.

**Comment by zxbodya** [comment]
if it is question about optimization - it should be possible to create that relation…

basically database structure is:
`object with galery --< gallery --< images`

and image url generation is not complicated.

---

however this was not considered to be primary usecase for extension(in most cases I remember - there was just one gallery or very rare cases 10s of them… which is not significant)

**Comment by Matvik** [comment]
In the case of, for example, an online store, there may be many products on one page, and for each product the main photo is displayed. Other variants also happen

**Comment by zxbodya** [comment]
yes, sure I do not argue there are no possibility for that, it just was not my case ;)

btw, for case with store - often for main photo, requirements are different (sizes of previews for example), and contexts where to use it also, are often different… so I have other extension for that - https://github.com/zxbodya/yii2-image-attachment
